zulipterminal/ui_tools/boxes.py

```diff
@@ -413,6 +413,10 @@ def footlinks_view(self,
                        ) -> Any:
         footlinks = []
         for link, (text, index) in message_links.items():
+            # Skip the footlink if its text only has the network location.
+            if text == urlparse(link).netloc:
+                continue
```

Comment:
I'm not sure if it's just this code/commit or the combination with previous commits - this works well for simple cases that I tested manually, but I think we may wish to be careful with handling the link scheme...or even the entire link?

Given a message of `example.com http://example.com https://example.com`, the webapp shows 3 links expressed as they are in the message (with 'text' exactly as written in the message), the first two pointing to the same address. We point the first two to the same address (based on the HTML, presumably), but we remove the scheme, so they *look* identical but are confusingly not, ie. we generate `example.com [1] example.com [1] example.com [2]`. Given we don't show the footlinks for these now after this commit, this is rather confusing! So, at a minimum, I think we should leave the scheme as it is, and probably use the assumed href in the anchor where no scheme is present?

After some further manual testing, note that even a long link like eg `example.com/blah/foooo/agikhasdglkashdglkashdg/asdfgklhasg/asdjkgalskjdhalskjdhlkasdjhsalkdghsasiougypoadisgypsdgypodsgypoadgysapdgy` gets shown twice identically (in message and footlinks). I don't think we gain from this being in the footlinks, much like with the simple `example.com` case? So, if there is no text, this makes me think we should remove it in all cases. Thoughts?

Reply:
Thanks for the report. It turns out that the server translates any scheme-less link to `HTTP` (e.g. `example.com` becomes `<a href="http://example.com">example.com</a>`). The core problem was with the `text == link` check that I had previously. I have replaced it with a 'scheme-less' check to figure out whether the 'text' part is identical.

Moreover, I have amended the last segment bit to only work when the text is not too simple (e.g. `example.com`) and to patch texts like these with their link itself. Consequently, `example.com http://example.com https://example.com` come out as `http://example.com [1] http://example.com [1] https://example.com [2]` and long links work more reliably.

Furthermore, I have not redacted the last segment bit for long links as I believe they improve the reading flow drastically. We could surely improve how we detect the 'significant' segment later but I think they work well for now.

Reply:
(See also https://github.com/zulip/zulip-terminal/pull/703/commits/a782d546470df674c78e7d52268fd89765b405f0#diff-3e518af2a88ffb14e1771a621641edd5R1552-R1558.)